### ISSUE TITLE
ES6+ linting

### ISSUE THREAD

**Issue by Extra-lightwill** [reactql/cli#25]
I have a component which uses ES6+ features e.g.

`myComponent extends React.Component { static propTypes = {`

This throws a syntax error as expected because it is an ES6+ feature. In the past I have installed the relevant npm modules e.g. ("babel-preset-stage-0":) and altered the babel.rc file and it has been successful but even with these additions in this setup, i'm still getting syntax errors:

My babel.rc file looks like this

```
{
  "presets": [
    ["env", {
      "targets": {
        "node": true
      }
    }],
    "react",
    "stage-0",
    "es2015"
  ]
}

```

Package.json:

```
"babel-preset-es2015": "^6.24.1",
"babel-preset-react": "^6.24.1",
"babel-preset-stage-0": "^6.24.1",

```

**Comment by hellojere** [comment]
I have a similar issue whilst trying to use ES7 decorators. I managed to get it running on development by adding  `transform-decorators-legacy` and `transform-class-properties` to `/kit/webpack/browser.js`, but couldn't figure out a way that would work for production as well. 

Having these in `.babelrc` didn't work at all. 

Any tips appreciated.

**Comment by leebenson** [comment]
Thanks for reporting. I'll add ES7 support to the next release, hopefully tomorrow.

`.babelrc` is solely to kick-start the Webpack build process; all code-level syntax is handled inside the `browser.js` and `server.js` Webpack config files in `kit/webpack`, so that's where plugins and transforms need to go.

Will report back here when decorator and static support is added.

**Comment by jtmthf** [comment]
I'm all for supporting some esnext features, but I would recommend only enabling specific transforms that are popular instead of setting stage-x presets. There's some talk among the Babel contributors on removing stage-x presets in the next version anyways due to the danger of their use and difficulty maintaining.

**Comment by leebenson** [comment]
Agreed @jtmthf - presets like `stage-0` are too broad, and would defeat the purpose of the more selective `env` preset. This enhancement would be specifically about adding static class properties and decorators.

**Comment by leebenson** [comment]
Moving this to https://github.com/reactql/kit/issues/2

**Comment by leebenson** [comment]
Decorators and static properties have been now added to the [kit](https://github.com/reactql/kit)

**Comment by hellojere** [comment]
Thanks a million! Off topic, but is there an official way to upgrade an existing project created with the CLI?

**Comment by leebenson** [comment]
@hellojere - my pleasure. Upgrading-- unfortunately not. Please see my [comment here](https://github.com/reactql/cli/issues/10#issuecomment-294115818) for some more thoughts on upgrading.

TL;DR- Since this is a starter kit rather than an imported framework, kit + custom code live side-by-side, making it tricky to upgrade without overwriting changes. The easiest way to upgrade is to create a new project, copy over `src` and then manually make any changes to the `kit` that may have changed between versions.
